components/common/RangeDatePicker/CustomDayContent.tsx

```diff
@@ -0,0 +1,105 @@
+import { isAfter, isBefore } from 'date-fns';
+import { DayContentPropsType } from './RangeDatePicker.types';
+
+function CustomDayContent(props: DayContentPropsType) {
+  const { date, range, isDayDisabled, currentDateHover } = props;
+  const todayDate = new Date().setHours(0, 0, 0, 0);
+
+  const hasRangeCellBackgroundHover = (date: Date) => {
+    if (!currentDateHover || !range?.from || range?.to) return false;
+
+    if (isAfter(date, range?.from) && isBefore(date, currentDateHover))
+      return true;
+
+    if (isBefore(date, range?.from) && isAfter(date, currentDateHover))
+      return true;
+
+    return false;
+  };
+
+  const hasCurrentCellBackgroundHover = (date: Date, isAfterFrom: boolean) => {
+    if (
+      range?.from &&
+      !range?.to &&
+      currentDateHover &&
+      date.getTime() === currentDateHover.getTime()
+    ) {
+      if (isAfterFrom) return isAfter(date, range.from);
+
+      return isBefore(date, range.from);
+    }
+
+    return false;
+  };
+
+  const hasFromCellBackgroundHover = (date: Date, isAfterCurrent: boolean) => {
+    if (
+      range?.from &&
+      !range.to &&
+      currentDateHover &&
+      date.getTime() === range?.from?.getTime()
+    ) {
+      if (isAfterCurrent) return isAfter(range.from, currentDateHover);
+
+      return isBefore(range.from, currentDateHover);
+    }
+
+    return false;
+  };
+
+  return (
+    <div
+      className={`aspect-square	 w-full h-auto my-5 ${
+        range?.from && range.to && date.getTime() === range?.from?.getTime()
+          ? 'bg-overlay rounded-s-full'
+          : ''
+      } ${
+        hasFromCellBackgroundHover(date, true)
+          ? 'bg-overlay  rounded-e-full'
+          : ''
+      } ${
+        hasFromCellBackgroundHover(date, false)
+          ? 'bg-overlay rounded-s-full'
+          : ''
+      } ${
+        range?.to && date.getTime() === range?.to?.getTime()
+          ? 'bg-overlay rounded-e-full'
+          : ''
+      } ${
+        hasCurrentCellBackgroundHover(date, true)
+          ? 'bg-overlay rounded-e-full'
+          : ''
+      } ${
+        hasCurrentCellBackgroundHover(date, false)
+          ? 'bg-overlay rounded-s-full'
+          : ''
+      } ${
+        range?.from &&
+        range.to &&
+        isAfter(date, range.from) &&
+        isBefore(date, range?.to)
```

Comment:
I believe it is preferable to include conditions such as these in a variable with the appropriate name.

Reply:
Fixed